### PR TITLE
Fix redo keyboard shortcut on macOS

### PR DESCRIPTION
**Pull request by aaronraimist**
I added the correct macOS shortcut for redo which is Cmd+Shift+Z. I had to reorder the if statement so redo is checked first, otherwise it would detect the undo first and never check that I was actually trying to redo.

Fixes https://github.com/vector-im/riot-web/issues/11298

### CODE REVIEW

File: src/components/views/rooms/BasicMessageComposer.js [hunk 1]
```diff
@@ -370,6 +370,16 @@ export default class BasicMessageEditor extends React.Component {
         } else if (modKey && event.key === Key.GREATER_THAN) {
             this._onFormatAction("quote");
             handled = true;
+        // redo
+        } else if ((modKey && event.key === Key.Y) ||
```
Comment (dbkr) [inline src/components/views/rooms/BasicMessageComposer.js]:
```suggestion
        } else if ((!IS_MAC && modKey && event.key === Key.Y) ||
```

File: src/components/views/rooms/BasicMessageComposer.js [hunk 2]
```diff
@@ -370,6 +370,16 @@ export default class BasicMessageEditor extends React.Component {
...
+                  (IS_MAC && event.shiftKey && event.key === Key.Z)) {
```
Comment (dbkr) [inline src/components/views/rooms/BasicMessageComposer.js]:
```suggestion
                  (IS_MAC && modkey && event.shiftKey && event.key === Key.Z)) {
```
...and looks like this isn't checking for cmd?

File: src/components/views/rooms/BasicMessageComposer.js (hunk 1 above)
Comment (dbkr) [reply src/components/views/rooms/BasicMessageComposer.js]: Probably should check !IS_MAC here so cmd+y doesn't get captured as redo on Mac

**Comment by t3chguy** [comment]
@aaronraimist can you get it passing CI please.

Thanks

**Comment by t3chguy** [comment]
can you try pulling in latest develop too, sorry about this

**Pull request merged**